databricks/koalas/generic.py

```diff
@@ -311,6 +313,70 @@ def abs(self):
         # TODO: The first example above should not have "Name: abs(0)".
         return _spark_col_apply(self, F.abs)
 
+    def clip(self, lower: Union[float, int] = None, upper: Union[float, int] = None) \
+            -> Union['ks.Series', 'ks.DataFrame']:
+        """
+        Trim values at input threshold(s).
+
+        Assigns values outside boundary to boundary values. Thresholds can be singular values or
+        array like, and in the latter case the clipping is performed element-wise in the specified
+        axis.
+
+        Parameters
+        ----------
+        lower : float or int, default None
+            Minimum threshold value. All values below this threshold will be set to it.
+        upper : float or int, default None
+            Maximum threshold value. All values above this threshold will be set to it.
+
+        Returns
+        -------
+        Series or DataFrame
+            Same type as calling object with the values outside the clip boundaries replaced
+
+        Examples
+        --------
+        >>> ks.Series([0, 2, 4]).clip(1, 3)
+        0    1
+        1    2
+        2    3
+        Name: 0, dtype: int64
+
+        >>> ks.DataFrame({'A': [0, 2, 4]}).clip(1, 3)
+           A
+        0  1
+        1  2
+        2  3
+
+        Notes
+        -----
+        One difference between this implementation and pandas is that running
+        pd.DataFrame({'A': ['a', 'b']}).clip(0, 1) will crash with "TypeError: '<=' not supported
+        between instances of 'str' and 'int'" while ks.DataFrame({'A': ['a', 'b']}).clip(0, 1)
+        will output the original DataFrame, simply ignoring the incompatible types.
+        """
+        if is_list_like(lower) or is_list_like(upper):
+            raise ValueError("List-like value are not supported for 'lower' and 'upper' at the " +
+                             "moment")
+
+        if lower is None and upper is None:
+            return self  # type: ignore
+
+        is_df = isinstance(self, ks.DataFrame)
+
+        df = self if is_df else self.to_dataframe()  # type: ignore
+        sdf = df._sdf
+
+        if lower is not None:
+            sdf = sdf.select(*(F.when(F.col(c) < lower, lower).otherwise(F.col(c)).alias(c)
```

Comment:
Can we optimize this a little bit to only apply this on numeric columns? for non-numeric columns, just keep the original column. This would make the compute much faster on real datasets with mixed type columns.

Reply:
With ac66cdc clipping is only applied to numerical columns now.